### PR TITLE
feat: add TLS support

### PR DESCRIPTION
**Pull request by Tochemey**
Basically add the TLS support by adding a simple TLSConfig to the HttpTransportOptions.

@thrawn01 I hope this was your plan.

### CODE REVIEW

File: transport/http_transport.go [hunk 1]
```diff
@@ -469,7 +502,16 @@ func waitForConnect(ctx context.Context, address string, cfg *tls.Config) error
 			errs = append(errs, ctx.Err().Error())
 			return errors.New(strings.Join(errs, "\n"))
 		}
-		time.Sleep(time.Millisecond * 100)
+		wait(time.Millisecond * 100)
```
Comment (thrawn01) [inline transport/http_transport.go:505]: Can you explain why `time.Sleep` isn't suitable here.

**Review by thrawn01 — APPROVED** [review]
Thank you for making this PR!

**Comment by thrawn01** [comment]
I noticed this PR doesn't include any client side TLS configuration. For example, If a TLS config requires a particular CA, then groupcache peers will be able to connect to that TLS enabled groupcache instance without proper client configuration.

**Comment by thrawn01** [comment]
I added a test for TLS here #6. Your PR forced me to finally publish `autotls` in `github.com/kapetan-io/tackle`, for which I have been procrastinating 😆 

Thanks for providing me with the motivation!

File: transport/http_transport.go (hunk 1 above)
Comment (Tochemey) [reply transport/http_transport.go:505]: @thrawn01 we can use sleep. This is just a preference.

Comment (thrawn01) [reply transport/http_transport.go:505]: Let's use sleep please.

**Pull request merged**